cloud_chemistry/prototype/include/aerosol/math.hpp

```
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cmath>
+
+namespace miam
+{
+
+/// @brief Calculate the effective radius for a log-normal aerosol mode
+/// @param mass Total mass concentration of particles in the mode [kg/m³]
+/// @param N Total number concentration of particles in the mode [#/m³]
+/// @param density Particle density [kg/m³]
+/// @param sig_g Geometric standard deviation (unitless)
+/// @return Effective radius [m]
+double CalculateEffectiveRadius(double mass, double N, double density, double sig_g) 
+{
+  if (N < 1e-10 || mass < 1e-18) return 0.0;
```

Do we know why these limits are set?

---

If the mass concentration is zero, then the radius is zero, which is physically inviable. Also N appears in the denominator, so it cannot be zero to avoid division by zero. But the decimal precision is chosen arbitrarily

---

Maybe instead of returning zero, we should either make these be asserts, or explicitly throw an error. If we are getting something that isn't meaningful it would be best to make it obvious, especially since these values don't have any physical meaning that we are aware of

---

Good point. I updated the function to throw the runtime error when the condition is met